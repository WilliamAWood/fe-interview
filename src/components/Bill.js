import React, {Component} from 'react';
import {StyledRow} from "../styled-components/StyledRow";
import {buttonStyle} from "../styled-components/buttonStyle";

class Bill extends Component {

    render() {

        const {bill, isActive, onButtonClick, onRowClick} = this.props;
        return (

            <li key={bill.id}>
                <StyledRow
                    data-test={`billRow-${bill.id}`}
                    isActive={isActive}
                    onClick ={() => onRowClick()}>
                    {bill.name}
                    {isActive &&
                    <ul>
                        {bill.transactions.map(transaction => (
                            <li key={transaction.id}>
                                <p>date: {transaction.date}</p>
                                <p>amount: {transaction.amount}</p>
                            </li>
                        ))}
                    </ul>}
                </StyledRow>
                <button style={buttonStyle} data-test={`billButton-${bill.id}`}
                        onClick={() => onButtonClick()}>
                    {bill.isBill ? 'is not bill' : 'is bill'}
                </button>
            </li>
        );
    }
}

export default Bill;