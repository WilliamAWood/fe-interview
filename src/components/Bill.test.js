
import React from 'react';
import renderer from 'react-test-renderer';
import Bill from "./Bill";

const props = {
    bill: {
        id: 1,
        isBill: true,
        name: "Vodafone",
        transactions: [
            {
                "amount": 12,
                "date": "2018-01-13",
                "id": 1
            }]
    },
    isActive: true,
};

it('renders correctly', () => {
    const tree = renderer
        .create(<Bill {...props}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
