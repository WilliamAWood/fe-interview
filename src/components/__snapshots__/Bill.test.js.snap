// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li>
  <div
    className="sc-bdVaJa fxcUYs"
    data-test="billRow-1"
    onClick={[Function]}
  >
    Vodafone
    <ul>
      <li>
        <p>
          date: 
          2018-01-13
        </p>
        <p>
          amount: 
          12
        </p>
      </li>
    </ul>
  </div>
  <button
    data-test="billButton-1"
    onClick={[Function]}
    style={
      Object {
        "background": "white",
        "border": "2px solid palevioletred",
        "color": "palevioletred",
      }
    }
  >
    is not bill
  </button>
</li>
`;
